lib/workers/repository/init/config.ts

```diff
@@ -220,9 +220,9 @@ export async function getRepoConfig(
 ): Promise<RenovateConfig> {
   let config = { ...config_ };
   config.baseBranch = config.defaultBranch;
-  config = await checkOnboardingBranch(config);
-  config = await mergeRenovateConfig(config);
   config.semanticCommits =
     config.semanticCommits ?? (await detectSemanticCommits());
+  config = await checkOnboardingBranch(config);
+  config = await mergeRenovateConfig(config);
```

Comment:
This reordering means that semantic commit detection will be done *before* `renovate.json` reading. Is that intentional? Will the user's config override it?